samples/js/fb/find-photos/src/main.js

```diff
@@ -143,6 +144,16 @@ async function getSizes(input): Promise<Map<Struct, string>> {
   return result;
 }
 
+async function getResources(input): Promise<Map<Struct, Struct>> {
+  let result = Promise.resolve(new Map());
+  await input.images.forEach(v => {
+    result = result.then(m => m.set(
```

Comment:
Since you've marked this function async, why not get rid of the promises inside the loop:

```
let m = new Map();
await input.images.forEach(v => {
  m = await m.set(...);
});
return m;
```